### PR TITLE
Story 8

### PR DESCRIPTION
**Pull request by JunePaloma**
#15 

Description of Changes Made: 
 Add feature test for story 8 (user authorization); add sad path testing to story 8 and story 9 (user can create account), added model test for the user, added validations and enums to user model; changed migration for users to included a default role. Change sessions create method in controller to redirect to the Dashboard_index_path instead of root. 

Anticipated Code Effects: 

User model and migrations, session controller 

Reviewer: 
@samuelssnider  @mimilettd  

props to @rickyamparo for navigation.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ def create
     if @user && @user.authenticate(params[:session][:password])
       session[:user_id] = @user.id
       flash[:notice] = "Logged in as #{@user.first_name} #{@user.last_name}"
-      redirect_to root_path
+      redirect_to dashboard_index_path
```
Comment (samuelssnider) [inline app/controllers/sessions_controller.rb]: This is the change we were talking about earlier, nice!

File: app/models/user.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
 class User < ApplicationRecord
   has_secure_password
+  validates :first_name, :last_name, :password, presence: true
+  validates :email, presence: true, uniqueness: true
+
+  enum role: ["default", "admin"]
```
Comment (samuelssnider) [inline app/models/user.rb]: Cool!

File: spec/features/user_authentication_spec.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          fill_in "session[email]", with: "testerson@testmail.com"
+          fill_in "session[password]", with: "testing"
+
+          click_on "Login"
+
+          expect(current_path).to eq(dashboard_index_path)
+          expect(page).to have_content "Logged in as Tester McTesty"
+          expect(page).to have_content "testerson@testmail.com"
+
+          expect(page).to_not have_link "Login"
```
Comment (samuelssnider) [inline spec/features/user_authentication_spec.rb]: I like doublechecking here, @rickyamparo , @JunePaloma .

File: spec/features/user_authentication_spec.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+            click_on "Login"
+            expect(current_path).to eq(login_path)
+
+            fill_in "session[email]", with: "testerson@testmail.com"
+
+
+            click_on "Login"
+
+            expect(current_path).to eq(login_path)
```
Comment (samuelssnider) [inline spec/features/user_authentication_spec.rb]: Good stuff, maybe for the refactor we can have situational flashes like 'no password entered'

File: spec/features/user_authentication_spec.rb [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+              visit '/'
+
+              expect(page).to have_link("Login")
+
+              click_on "Login"
+              expect(current_path).to eq(login_path)
+
+              fill_in "session[email]", with: "tester@testmail.com"
+              fill_in "session[password]", with: "testing"
```
Comment (samuelssnider) [inline spec/features/user_authentication_spec.rb]: Thorough, nice!

File: spec/features/user_authentication_spec.rb (hunk 4 above)
Comment (rickyamparo) [reply spec/features/user_authentication_spec.rb]: Absolutely! I like the idea of having flashes dependent on the situation.

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  context "invalid attributes" do
+      it "is invalid without a first name" do
+        user = User.new(last_name: "McTest", password: "testing", email: "tester@testmail")
+        expect(user).to be_invalid
+      end
+      it "is invalid without a last name" do
+        user1 = User.new(first_name: "McTest", password: "testing", email: "tester@testmail")
+        expect(user1).to be_invalid
+      end
+      it "is invalide without a password" do
```
Comment (samuelssnider) [inline spec/models/user_spec.rb]: typo on invalid

**Pull request merged**